File: python-wrapper/examples/multi_agent.py
import asyncio
import os
from ferrant import Agent, Team

async def main() -> None:
    key = os.environ["OPENAI_API_KEY"]
    researcher = Agent.openai("gpt-5-nano", key, instructions="Identify facts and uncertainty.")
    reviewer = Agent.openai("gpt-5-nano", key, instructions="Find risks and edge cases.")
    team = Team.openai("gpt-5-nano", key, [
        ("researcher", "Researches the subject", researcher),
        ("reviewer", "Critically reviews proposals", reviewer),
    ])
    print(await team.run("Propose a safe VM-to-containers migration and consult both specialists."))

asyncio.run(main())

File: examples/multi_agent.rs
//! Coordinator-led team with two specialist agents.

use ferrant::llm::openai::OpenAiModel;
use ferrant::{Agent, AgentTeam};

fn model(api_key: &str) -> OpenAiModel {
    OpenAiModel::new("gpt-5-nano", api_key)
}

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    dotenvy::dotenv().ok();
    let key = std::env::var("OPENAI_API_KEY")?;

    let researcher = Agent::builder(model(&key))
        .instructions(
            "You are a careful research analyst. Identify facts, assumptions, and uncertainty.",
        )
        .build();
    let reviewer = Agent::builder(model(&key))
        .instructions(
            "You are a critical reviewer. Find risks, edge cases, and practical improvements.",
        )
        .build();

    let mut team = AgentTeam::new(model(&key))
        .member(
            "researcher",
            "Researches and analyzes the subject",
            researcher,
        )
        .member(
            "reviewer",
            "Critically reviews a proposal and finds risks",
            reviewer,
        );

    let answer = team.run(
        "Propose a safe rollout plan for moving a small API from a VM to containers. Ask both specialists and synthesize their advice."
    ).await?;
    println!("{answer}");
    Ok(())
}
